src/Hl7.Fhir.Specification/Specification/Source/WebResolver.cs

```diff
@@ -24,25 +24,25 @@ public class WebResolver : IResourceResolver, IAsyncResourceResolver
         /// <summary>Default request timeout in milliseconds.</summary>
         public const int DefaultTimeOut = 5000;
 
-        readonly Func<Uri, FhirClient> _clientFactory;
+        readonly Func<Uri, BaseFhirClient> _clientFactory;
 
         /// <summary>Default constructor.</summary>
         public WebResolver() { }
 
-        /// <summary>Create a new <see cref="WebResolver"/> instance that supports a custom <see cref="FhirClient"/> implementation.</summary>
+        /// <summary>Create a new <see cref="WebResolver"/> instance that supports a custom <see cref="BaseFhirClient"/> implementation.</summary>
         /// <param name="fhirClientFactory">
-        /// Factory function that should create a new <see cref="FhirClient"/> instance for the specified <see cref="Uri"/>.
-        /// If this parameter equals <c>null</c>, then the new instance creates a default <see cref="FhirClient"/> instance.
+        /// Factory function that should create a new <see cref="BaseFhirClient"/> instance for the specified <see cref="Uri"/>.
+        /// If this parameter equals <c>null</c>, then the new instance creates a default <see cref="BaseFhirClient"/> instance.
         /// </param>
-        public WebResolver(Func<Uri, FhirClient> fhirClientFactory)
+        public WebResolver(Func<Uri, BaseFhirClient> fhirClientFactory)
```

Comment:
This is a change in the public interface.

Reply:
Yes indeed. We can also decide to create a common `WebResolver` and move this `WebResolver` as a shim back to the satellites.

Reply:
I've created a `CommonWebSource`.